task7/script.js

```diff
@@ -0,0 +1,369 @@
+            name: "Artsiom Kudryavtsev",
+            photoLink: "zZuTNe0PW34.jpg",
+            hashtags: [],
+            likes: ["ar_kud"]
+        },
+        {
+            id: '17',
+            description: "Как же хорошо, что у меня наконец появилось время подтянуть все свои долги!",
+            createdAt: new Date('2020-05-03T17:31:24'),
+            author: "ar_kud",
+            name: "Artsiom Kudryavtsev",
+            photoLink: "zZuTNe0PW34.jpg",
+            hashtags: [],
+            likes: ["ar_kud"]
+        },
+        {
+            id: '18',
+            description: "Как же хорошо, что у меня наконец появилось время подтянуть все свои долги!",
+            createdAt: new Date('2020-05-03T17:31:30'),
+            author: "ar_kud",
+            name: "Artsiom Kudryavtsev",
+            photoLink: "zZuTNe0PW34.jpg",
+            hashtags: [],
+            likes: ["ar_kud"]
+        },
+        {
+            id: '19',
+            description: "Как же хорошо, что у меня наконец появилось время подтянуть все свои долги!",
+            createdAt: new Date('2019-06-03T17:31:24'),
+            author: "ar_kud",
+            name: "Artsiom Kudryavtsev",
+            photoLink: "zZuTNe0PW34.jpg",
+            hashtags: [],
+            likes: ["ar_kud"]
+        },
+        {
+            id: '20',
+            description: "Как же хорошо, что у меня наконец появилось время подтянуть все свои долги!",
+            createdAt: new Date('2020-05-01T17:31:24'),
+            author: "ar_kud",
+            name: "Artsiom Kudryavtsev",
+            photoLink: "zZuTNe0PW34.jpg",
+            hashtags: [],
+            likes: ["ar_kud"]
+        }
+    ];
+
+    class ChirikList {
+
+        constructor(posts) {
+            this._posts = [];
+            this._deletedPosts = [];
+            posts.forEach((post) => {
+                if (ChirikList.validatePost(post)) {
+                    ChirikList.updateHashtags(post);
+                    this._posts.push(post);
+                }
+            });
+            console.log("New ChirikList object with " + this._posts.size + " elements");
+        }
+
+        addAll(posts) {
+            let notValidPosts;
+            posts.forEach((post) => {
+                if (ChirikList.validatePost(post)) {
+                    ChirikList.updateHashtags(post);
+                    this._posts.push(post);
+                } else
+                    notValidPosts.push(post);
+            });
+            console.log("Added " + this._posts.size + " elements, " + notValidPosts.size + " elements not valid");
+
+            return notValidPosts;
+        }
+
+        getPosts(skip, top, filterConfig) {
+            skip = skip || 0;
+            top = top || 10;
+
+            let result = this._posts.slice();
+
+            if (filterConfig.hashtags) {
+                result = result.filter(function (item) {
+                    if (item.hashtags === null || item.hashtags === undefined)
```

Comment:
It's better to check direct type
e.g.  (Array.isArray(item.hashtags)),  (item.date instanceof Date)

Reply:
Done.